test/unittest/blas1_axpy_test.cpp

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_axpy_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas1_test.hpp"
+
+typedef ::testing::Types<blas1_test_args<float>, blas1_test_args<double> >
+    BlasTypes;
+
+TYPED_TEST_CASE(BLAS1_Test, BlasTypes);
+
+REGISTER_SIZE(::RANDOM_SIZE, axpy_test)
+REGISTER_STRD(::RANDOM_STRD, axpy_test)
+REGISTER_PREC(float, 1e-4, axpy_test)
+REGISTER_PREC(double, 1e-6, axpy_test)
+REGISTER_PREC(long double, 1e-7, axpy_test)
```

Comment:
It probably should be higher, meaning I have to reduce random sizes as well. But relative error easily runs away, so I would not stick to that.